### PR TITLE
fix: start project pipeline update gittar token and check permission

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
There are several important defects in the project pipeline that need to be repaired, otherwise it will affect the use

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb24iOls4ODMsNzcyXSwibWVtYmVyIjpbIjEwMDEyNjEiXX0%3D&id=275735&iterationID=772&pId=0&type=BUG)

- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb24iOls4ODMsNzcyXSwibWVtYmVyIjpbIjEwMDEyNjEiXX0%3D&id=275759&iterationID=772&pId=0&type=BUG)

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       The pipeline of the newly created feature branch of application development reports an error, and there is no restriction on the deletion of the running pipeline       |
| 🇨🇳 中文    |      应用开发新建的feature分支的流水线，执行报错，运行中的流水线删除没有限制        |

### CODE REVIEW

File: modules/dop/providers/projectpipeline/service_impl.go [hunk 1]
```diff
@@ -245,14 +248,37 @@ func (p *ProjectPipelineService) Delete(ctx context.Context, params deftype.Proj
 		return nil, apierrors.ErrDeleteProjectPipeline.InvalidParameter(err)
 	}
 
-	_, source, err := p.getPipelineDefinitionAndSource(params.ID)
+	definition, source, err := p.getPipelineDefinitionAndSource(params.ID)
 	if err != nil {
 		return nil, apierrors.ErrDeleteProjectPipeline.InvalidParameter(err)
 	}
+	if definition.Creator != apis.GetUserID(ctx) {
```
Comment (sfwn) [inline modules/dop/providers/projectpipeline/service_impl.go:255]: If person resigned, this pipeline can never be deleted?

File: modules/dop/providers/projectpipeline/service_impl.go [hunk 2]
```diff
@@ -245,14 +248,37 @@ func (p *ProjectPipelineService) Delete(ctx context.Context, params deftype.Proj
...
+		Sources:  []apistructs.PipelineSource{extraValue.CreateRequest.PipelineSource},
+		YmlNames: []string{extraValue.CreateRequest.PipelineYmlName},
+		PageSize: 1,
+		PageNo:   1,
+	})
+	if err != nil {
+		return nil, apierrors.ErrDeleteProjectPipeline.InternalError(err)
+	}
+	if len(crons.Data) > 0 && crons.Data[0].Enable != nil && *crons.Data[0].Enable == true {
+		return nil, apierrors.ErrDeleteProjectPipeline.InternalError(fmt.Errorf("pipeline cron was running status"))
```
Comment (sfwn) [inline modules/dop/providers/projectpipeline/service_impl.go:280]: is running status

File: modules/dop/providers/projectpipeline/service_impl.go (hunk 1 above)
Comment (kakj-go) [reply modules/dop/providers/projectpipeline/service_impl.go:255]: For the time being, do permission management later

Comment (sfwn) [reply modules/dop/providers/projectpipeline/service_impl.go:255]: add TODO please

**Review by sfwn — APPROVED** [review]

**Pull request merged**